### ISSUE TITLE
Permission denied message when running as non-root user

### ISSUE THREAD

**Issue by bilogic** [laravel/pint#141]
- Pint Version: 1.4.0
- OS version: Ubuntu 22.04.1 LTS
- PHP version: 8.1.14
- Laravel version: 8.83.27

### Description:
```
$ /usr/local/composer/vendor/bin/pint index.php

  .
In Filesystem.php line 190:
                                                                                                                  
  file_put_contents(/tmp/2cfc4aa12cd8f861cd82500f40525531b37ea16e.php): Failed to open stream: Permission denied  

$                                                                                                                  
```
Pint is installed globally in my system. Above is my output. It seems strange that there is an additional permission denied message (Not exception) when I run as non-root user. However, styling issues in `index.php` was fixed.

### Steps To Reproduce:
Run pint on any PHP file using a non-root user

**Comment by nunomaduro** [comment]
Seems that, at some point, you installed Pint in your system using `sudo`. Meaning that, now, non-sudo users in your system wont have permission to run Pint. Do you mind of removing Pint, and re-install it, this time not using sudo?

**Comment by Colonel-Sandvich** [comment]
@bilogic 
To me it looks like you're trying to write to `/tmp/...` which is a protected folder and you would need `sudo` to write to this. Nothing to do with pint, just a file permissions problem.

**Comment by bilogic** [comment]
Hmm, let me rephrase the whole issue.

1. I installed pint globally and tried to fix 1 index.php which is not in `/tmp/...`
2. It fixed the index.php but gave a message (not exception)
3. Nuno's suggestion to re-install made the message go away
4. However, in pt2, `index.php` was already styled, so why was pint still trying to write to `/tmp/...`? Is it necessary?

> you're trying to write to `/tmp/...` which is a protected folder
5. As detailed above, my code is not the one trying to write to `/tmp/...`

pt4 would be my focus if there is interest in this issue

**Comment by Colonel-Sandvich** [comment]
Oh I see. My guess is that either pint or php-cs-fixer is storing some kind of cache. Since you installed it with `sudo` you'd have to keep running it with `sudo` to allow it to write to `/tmp`.

There's no issue here just don't install it with `sudo`; install per project

**Comment by bilogic** [comment]
@nunomaduro 

```
user3@linuxdev:~/project$ /usr/local/composer/vendor/bin/pint /home/user3/project/file_to_format.php

  .
In Filesystem.php line 204:
                                                                                                          
  file_put_contents(/tmp/9166d95859c7d6216febc6f920afe3cf.php): Failed to open stream: Permission denied  
                                                                                                          

user3@linuxdev:~/project$ 

```
```
drwxrwxrwt  16 root  root       36864 Apr  3 08:27 tmp/
```

I'm facing this again. 

1. I installed pint globally using `user3`, but when I run pint as `user3`, I get the above `Permission denied`.
2. Further, as mentioned last time, the `.` already indicates pint is done with `file_to_format.php`, why does it still need to write to `tmp` folder? `file_to_format.php` is owned by `user3` and writable.

Appreciate your insights. Thank you.

**Comment by bilogic** [comment]
@driesvints @nunomaduro 

Further observations:
```
<div class="mt-2 mx-2">
    <div class="flex space-x-1">
        <span class="flex-1 content-repeat-[─] text-gray">

        </span>

        <span class="text-gray">
            <?php echo e($preset); ?>

        </span>
    </div>
...
<?php /**PATH phar:///usr/local/composer/vendor/laravel/pint/builds/pint/resources/views/summary.blade.php ENDPATH**/ ?>
```

I run pint on 1 system as several users, 

1. After `user1` runs it, `/tmp/9166d95859c7d6216febc6f920afe3cf.php` is created and owned by `user1` and not deleted after pint exits.
2. When `user3`'s pint runs, it also tries to write to the exact same temp filename (`9166d95859c7d6216febc6f920afe3cf.php`) but throws the error because the file is owned by `user1`.
3. The impact with this error is my script can't differentiate between a fail formatting vs a actual exception
4. Can we have an option to do away with the summary blade or something? 

Thank you.

**Comment by takuyaw-w** [comment]
Pint Version: 1.0.0
OS version: Manjaro Linux x86_64
PHP version: 8.2.8
Laravel version: 10.10

'Laravel sail' was used to build the environment and I encountered the same phenomenon.
I ran the following and now the `sail pint` command works correctly.

```sh
sail root-shell
chown -R sail:sail /tmp
```

**Comment by Morinohtar** [comment]
I had the same problem. No idea why Pint created that tmp file as root, i dont remember ever running the command using root.
Either way, an easy way to solve it, which doesnt implicate reinstalling Pint or changing the `/tmp` folder ownership, is to just:

1. enter the shell as root:

    `sail root-shell`

2. delete the file in question:

    ex: 9166d95859c7d6216febc6f920afe3cf.php

    `rm /tmp/9166d95859c7d6216febc6f920afe3cf.php`

3. exit the shell and rereun Pint

    Pint will create the file again, but this time the owner will be the sail user.
